src/azure/cli/commands/arm.py

```diff
@@ -394,12 +402,25 @@ def _update_instance(instance, part, path):
     try:
         index = index_regex.match(part)
         if index:
-            try:
-                index_value = int(index.group(1))
-                instance = instance[index_value]
-            except IndexError:
-                raise CLIError('index {} doesn\'t exist on {}'.format(index_value,
-                                                                      make_camel_case(path[-2])))
+            if '=' in index.group(1):
+                key, value = index.group(1).split('=')
+                matches = [x for x in instance if isinstance(x, dict) and x.get(key, None) == value]
```

Comment:
The intent here wasn't to match empty string. It would be very odd to have the key be empty string.

{
  '': 'blah'
}

The purpose was to make sure that they wouldn't match if the key wasn't found. 

---
In reply to: [76155025](https://github.com/Azure/azure-cli/pull/740#discussion_r76155025) [](ancestors = 76155025)